src/views/statusBar.ts

```diff
@@ -0,0 +1,56 @@
+import * as vscode from 'vscode';
+import { getActiveStack } from '../commands/stackCommands';
+
+export class ZenMLStatusBar {
+  private static instance: ZenMLStatusBar;
+  private statusBar: vscode.StatusBarItem;
+  private constructor() {
+    this.statusBar = vscode.window.createStatusBarItem(vscode.StatusBarAlignment.Right, 100);
+    this.statusBar.command = 'zenml.showActiveStack';
+    this.updateStatusBar();
+  }
+
+  public static getInstance(): ZenMLStatusBar {
+    if (!ZenMLStatusBar.instance) {
+      ZenMLStatusBar.instance = new ZenMLStatusBar();
+      // ZenMLStatusBar.instance.startAutoRefresh();
+    }
+    return ZenMLStatusBar.instance;
+  }
+
+  // public startAutoRefresh() {
+  //   const interval = 30000;
+  //   setInterval(() => {
+  //     this.updateStatusBar();
+  //   }, interval);
+  // }
+
+  public show() {
+    this.statusBar.show();
+  }
+
+  public hide() {
+    this.statusBar.hide();
+  }
+
+  public updateStatusBar() {
+    console.log('Updating ZenML active stack...');
+
+    getActiveStack().then((activeStackCliOutput) => {
+      /**
+       * The cli command `zenml stack get` outputs the line below: 
+       * The global active stack is: 'default'
+       * 'default' is the actual string we want to display in the status bar.
+       */
+      const match = activeStackCliOutput.match(/'([^']+)'/); // matches text within single quotes
```

Comment:
How resilient is this regex? I think there might be an issue if a stack is named something like `zenml's stack` and contains a `'`.

Reply:
Good catch! Just adjusted the helper function to retrieve everything after `The global active stack is: `, and then to slice the first and and last character out of the string (see updated demo above).